stride/problem/geometry.py

```diff
@@ -283,6 +285,71 @@ def default(self, geometry_type, *args, **kwargs):
 
             self.add(index, self._transducers.get(0), _coordinates)
 
+    def from_fullwave(self, geompgy, **kwargs):
+        """
+        Populates geometry container based from a Fullwave geometry pgy file
+
+        Parameters
+        ----------
+        geompgy : Path
+            Path to .pgy Fullwave geometry file
+
+        scale: float, optional
+            Value to each scale the location values in all dimensions. Useful for unit conversion. Default 1.
+
+        disp: tuple or float, optional
+            Amount to displace in each dimension. Applied after scale. Default (0., 0., 0.)
+
+        dropdims: tuple or int, optional
+            Coordinate dimensions of .pgy file to drop (count from 0). Default ()
+
+        Returns
+        -------
+        """
+        assert geompgy.lower().split(".")[-1] == "pgy", "Expected .pgy extension but found .%s"%geompgy.lower().split(".")[-1]
+
+        num_locations, n1, n2, n3 = -1, -1, -1, -1
+        scale = kwargs.get('scale', 1.)
+        disp = kwargs.get('disp', (0., 0., 0.))
+        dropdims = kwargs.get('dropdims', ())
+
+        # Read coordinates and IDs from pgy file
+        with open(geompgy, 'r') as f:
+            for i, line in enumerate(f):
+                line = line.split()
+                
+                # Header line
+                if i == 0:
+                    num_locations, n1, n2, n3 = [int(h) for h in line] # nz, ny, nx in fullwave format
+                    coordinates = np.zeros((num_locations, len(line)-1))
```

Comment:
I'm a bit confused, shouldn't the coordinates be of size (num_locations, self.space.dim)?

Reply:
ideally yes, but fullwave, as far as I'm aware, always passes the three coordinates to the file. Restricting it to the dims of the problem might throw errors later on. I thought getting the length of the line would create consistency with the file created, but let me know if you still prefer to have self.space.dim